core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerBindingService.java

```diff
@@ -43,35 +46,47 @@
   private final ConsumerBindingRepository consumerBindingRepository;
   private final NotificationEventEmitter<ConsumerBinding> consumerBindingServiceEventEmitter;
 
-  public Optional<ConsumerBinding> create(ConsumerBinding consumerBinding) throws ValidationException {
-    if (consumerBindingRepository.findById(consumerBinding.getKey()).isPresent()) {
+  public Optional<ConsumerBinding> create(ConsumerBinding consumerBindingModel) throws ValidationException {
+    consumerBindingValidator.validateForCreate(consumerBindingModel);
+    var consumerBindingData = convertToData(consumerBindingModel);
+    if (consumerBindingRepository.findById(consumerBindingData.getKey()).isPresent()) {
       throw new ValidationException("Can't create because it already exists");
     }
-    consumerBindingValidator.validateForCreate(consumerBinding);
-    consumerBinding.setSpecification(handlerService.handleInsert(consumerBinding));
-    return consumerBindingServiceEventEmitter.emitEventOnProcessedEntity(EventType.CREATE, consumerBindingRepository.save(consumerBinding));
+    consumerBindingData.setSpecification(convertToData(handlerService.handleInsert(consumerBindingModel)));
+    ConsumerBinding out = convertToModel(consumerBindingRepository.save(consumerBindingData));
+    consumerBindingServiceEventEmitter.emitEventOnProcessedEntity(EventType.CREATE, out);
+    return Optional.ofNullable(out);
   }
 
   public Optional<ConsumerBinding> read(ConsumerBindingKey key) {
-    return consumerBindingRepository.findById(key);
+    var data = consumerBindingRepository.findById(convertToData(key));
+    return data.isPresent() ? Optional.of(convertToModel(data.get())) : Optional.empty();
   }
 
   public Iterable<ConsumerBinding> readAll() {
-    return consumerBindingRepository.findAll();
+    ArrayList out = new ArrayList();
```

Comment:
`Iterable` is a pain in the java8+ world...

FYI, since the switch from `CrudRepository` to `JpaRepository`, `findAll` now returns `List` so you could go streamy here (I assumed being `Iterable` before is the reason you've gone with the classic for loop

Reply:
Agreed, but returning iterable is. the existing functionality.

Reply:
`Iterable` was only chosen before because the Repository exposed it. We now have `List` exposed. Everywhere else is very streamy in style. This is the only place it is not.

Reply:
Changed to a List in all services.
Also removed unused 'findAll' methods